levels/level_manager.gd

```diff
@@ -0,0 +1,54 @@
+extends Node
+
+
+var level_data: Array = []
+
+
+@onready var spawn_timer = $SpawnTimer
+
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	_load_level(1)
+	
+	
+func _load_level(level_number: int):
+	
+	var file = FileAccess.open("res://levels/level%s.csv" % level_number, FileAccess.READ)
+	var level_data = _parse_level_file(file)
+
+	_progress_level()
+	
+func _parse_level_file(file: FileAccess):
+
+	# skip header
+	file.get_csv_line()
+	while true:
+		var line = file.get_csv_line()
+		# if we're on the last line then we can finish loading.
+		if len(line) == 1:
+			break
+		level_data.append(line)
+	
+	return level_data
+
+
+func _progress_level():
+	var next = level_data.pop_front()
+
+	print("spawn next step in level now!!")
+	print('step data: ', next)
+
+	var timeout = float(next[0])
+	var spawn_node = next[1]
+	var enemy = next[2]
```

Comment:
I wonder if we can't read the level data in to a class so that its typed & named? Like a python dataclass